src/buffer.rs

```diff
@@ -388,12 +395,22 @@ impl<'a> Eq for Cursor<'a> {}
 
 impl<'a> PartialEq for Cursor<'a> {
     fn eq(&self, other: &Self) -> bool {
-        let Cursor { ptr, scope, marker } = self;
-        let _ = marker;
+        let Cursor { ptr, scope, .. } = self;
         *ptr == other.ptr && *scope == other.scope
     }
 }
 
+impl<'a> PartialOrd for Cursor<'a> {
+    fn partial_cmp(&self, other: &Self) -> Option<cmp::Ordering> {
+        let Cursor { index, scope, .. } = self;
+        if *scope == other.scope {
+            Some(index.cmp(&other.index))
```

Comment:
(It does look odd to `PartialEq` over `(ptr, scope)` and `PartialOrd` over `(index, scope)`, but this should be consistent.)